storage/invoke.go

```diff
@@ -63,21 +63,30 @@ func run(ctx context.Context, call func() error, retry *retryConfig, isIdempoten
 	}
 
 	return internal.Retry(ctx, bo, func() (stop bool, err error) {
-		setRetryHeader(req, invocationID, attempts)
+		setHeader(invocationID, attempts)
 		err = call()
 		attempts++
 		return !errorFunc(err), err
 	})
 }
 
-func setRetryHeader(req interface{ Header() http.Header }, invocationID string, attempts int) {
-	if req == nil {
+func setRetryHeaderHTTP(req interface{ Header() http.Header }) func(string, int) {
+	return func(invocationID string, attempts int) {
+		if req == nil {
+			return
+		}
+		header := req.Header()
+		invocationHeader := fmt.Sprintf("gccl-invocation-id/%v gccl-attempt-count/%v", invocationID, attempts)
+		xGoogHeader := strings.Join([]string{invocationHeader, xGoogDefaultHeader}, " ")
+		header.Set("x-goog-api-client", xGoogHeader)
```

Comment:
I know that we "own" the x-goog-api-client header, but is there a way for this to just _append_ the invocation k-v-pairs to it rather than _set_ it and also be the function that sets the `xGoogDefaultHeader`. If any other part of the stack wants to modify this header to add more info to it, things get complicated.

Reply:
Discussed offline -- it's very difficult to do this with code that retries without appending the same information to the header multiple times. Seems best to take this approach for now.